### DISCUSSION TITLE
Help me set up my search Ajax

### DISCUSSION THREAD

**Discussion by bilalovmc** [aminalaee/sqladmin#632] [Q&A]
Tell me how to configure Ajax to add new data.
Perhaps there is a connection problem?

At the moment, everything is displayed well; you can delete an element, but you can’t add it; you can’t find it.

![image](https://github.com/aminalaee/sqladmin/assets/96948054/4c5d334b-9ab1-4c9d-b9e7-9db498058b26)

```
class Upselling(Base):
    __tablename__ = "upselling"

    id = Column(Integer, primary_key=True)  # type: ignore
    name = Column(String, unique=True)
       
    subsystem = relationship("SubSystemConn")
    

class UpsellingAdmin(ModelView, model=Upselling):
    icon = "fa-solid fa-basket-shopping"
    column_list = __all__
    form_excluded_columns = [Upselling.tasks]
    
    form_ajax_refs = {
        "subsystem": {
            "fields": ("name",),
        },
    }

class SubSystemConn(Base):
    __tablename__ = "subsystem_upselling"

    id = Column(Integer, primary_key=True)
    name = Column(Enum(SubSystem, values_callable=lambda en: [e.value for e in en]))
    upselling_id = Column(Integer, ForeignKey("upselling.id", ondelete="SET NULL"))

    __table_args__ = (UniqueConstraint("upselling_id", "name", name="upselling_subsystem"),)

    def __str__(self):
        return self.name.value

    def __repr__(self):
        return self.name.value
```

**Comment by aminalaee** [comment]
What is the SubSystem enum class?

**Reply by bilalovmc** [reply]
```
class SubSystem(str, enum.Enum):
    SYSTEM_1C = "1С"
    WEB = "WEB"
    CONNECTORS = "Коннектор"
    API = "API"
    UTILITY = "Утилита"
    NO_SUBSYSTEM = "No subsystem"
    PACKER_WORKSTATION = "АРМ Фасовщика"
    WORKLFLOW = "workflow"
    DOCUMENT_FLOW = "Документооборот"
    FORM = "Формы"
```

![image](https://github.com/aminalaee/sqladmin/assets/96948054/4d6246ce-6ced-4a8b-bd5e-7edcc89c58f5)

**Reply by aminalaee** [reply]
It works fine for me.
Do you have any `SubSystemConn` records inserted before being able to insert `Upselling`?

**Comment by bilalovmc** [comment]
<img width="638" alt="image" src="https://github.com/aminalaee/sqladmin/assets/96948054/eb90ffbd-78f3-46dd-872b-7e364c1aa6a3">

I discovered why it doesn't work; requests go to the wrong address. How can I change the path? I just have 127.0.0.1:5000

**Reply by bilalovmc** [reply]
base_url="/"   breaks Ajax 
```
admin = Admin(
        app,
        engine=engine,
        title="Upselling",
        base_url="/",
        authentication_backend=authentication_backend,
    )  
```
![image](https://github.com/aminalaee/sqladmin/assets/96948054/64a23572-e8d8-4e8e-8e42-90107d7c4a60)

Everything works after changing base_url

**Reply by aminalaee** [reply]
Fixed in https://github.com/aminalaee/sqladmin/pull/638 will be released soon
